synapse/handlers/room_member.py

```diff
@@ -872,9 +872,23 @@ def _ask_id_server_for_third_party_invite(
             "sender_avatar_url": inviter_avatar_url,
         }
 
-        data = yield self.simple_http_client.post_urlencoded_get_json(
-            is_url, invite_config
-        )
+        try:
+            data = yield self.simple_http_client.post_json_get_json(
+                is_url, invite_config
+            )
+        except HttpResponseException as e:
```

Comment:
can we restrict this to only happen on a 400 response?

Reply:
If we restrict it then we need to restrict it to 500s, because https://github.com/matrix-org/sydent/pull/170 fixes it 500ing on a JSON request.

Reply:
good point, ignore me